internal/job/checkout.go

```diff
@@ -787,6 +787,14 @@ const CommitMetadataKey = "buildkite:git:commit"
 // note that we bail early if the key already exists, as we don't want to overwrite it
 func (e *Executor) sendCommitToBuildkite(ctx context.Context) error {
 	e.shell.Commentf("Checking to see if git commit information needs to be sent to Buildkite...")
+
+	commitResolved, _ := e.shell.Env.Get("BUILDKITE_COMMIT_RESOLVED")
+	if commitResolved == "true" {
+		// we can skip the metadata shenanigans here and push straight through
+		e.shell.Commentf("BUILDKITE_COMMIT is already resolved and meta-data populated, skipping...")
```

Comment:
This doesn't have a follow-on comment, so I'd avoid the trailing dots:

```suggestion
		e.shell.Commentf("BUILDKITE_COMMIT is already resolved and meta-data populated, skipping")
```

Reply:
Good point! Will update alongside adding tests